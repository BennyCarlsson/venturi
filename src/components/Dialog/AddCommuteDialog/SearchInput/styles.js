import styled from 'styled-components';

export const Input = styled.input`
  width: 100%;
  height: 36px;
  border: none;
  border-bottom: 1px solid #726a95;
  font-family: 'Montserrat';
  font-style: normal;
  font-size: 14px;
  line-height: 17px;
  padding: 8px;
`;
