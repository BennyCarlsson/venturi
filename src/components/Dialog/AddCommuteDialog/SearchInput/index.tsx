import { Dispatch, SetStateAction } from "react";
import { Input } from "./styles";

type SearchInputProps = {
  value: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  setShowDropDown: Dispatch<SetStateAction<boolean>>;
};

const SearchInput = ({
  value,
  onChange,
  setShowDropDown,
  ...rest
}: SearchInputProps) => {
  return (
    <Input
      {...rest}
      value={value}
      onChange={onChange}
      onBlur={() => {
        setTimeout(() => {
          setShowDropDown(false);
        }, 100);
      }}
      onFocus={() => {
        if (value.length > 1) {
          setShowDropDown(true);
        }
      }}
    />
  );
};

export default SearchInput;
