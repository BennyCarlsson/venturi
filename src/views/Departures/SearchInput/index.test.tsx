import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SearchInput from '.';

describe('<SearchInput />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<SearchInput title="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    render(<SearchInput title="test" />);
    expect(screen).toBeTruthy();
  });
});
