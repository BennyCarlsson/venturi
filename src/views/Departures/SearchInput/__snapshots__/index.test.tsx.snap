// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapshot 1`] = `
<input
  className="sc-bdfBwQ kDyVWm"
  onBlur={[Function]}
  onFocus={[Function]}
/>
`;
